src/acl.c

```diff
@@ -1590,7 +1703,8 @@ void addACLLogEntry(client *c, int reason, int keypos, sds username) {
 
     switch(reason) {
     case ACL_DENIED_CMD: le->object = sdsnew(c->cmd->name); break;
-    case ACL_DENIED_KEY: le->object = sdsnew(c->argv[keypos]->ptr); break;
+    case ACL_DENIED_KEY: le->object = sdsnew(c->argv[argpos]->ptr); break;
+    case ACL_DENIED_CHANNEL: le->object = sdsnew(c->argv[argpos]->ptr); break;
     case ACL_DENIED_AUTH: le->object = sdsnew(c->argv[0]->ptr); break;
```

Comment:
optimization: we can use sdsnewlen, or sdsdup

Reply:
Done.